arbor/include/arbor/schedule.hpp

```diff
@@ -1,190 +1,39 @@
 #pragma once
 
 #include <algorithm>
-#include <iterator>
 #include <memory>
-#include <random>
 #include <type_traits>
 #include <utility>
 #include <vector>
+#include <random>
 
 #include <arbor/assert.hpp>
 #include <arbor/common_types.hpp>
 #include <arbor/util/extra_traits.hpp>
 #include <arbor/export.hpp>
 #include <arbor/serdes.hpp>
+#include <arbor/units.hpp>
 
 // Time schedules for probe–sampler associations.
 namespace arb {
 
+using engine_type = std::mt19937_64;
+using seed_type = std::remove_cv_t<decltype(engine_type::default_seed)>;
+
+constexpr static auto default_seed = engine_type::default_seed;
```

Comment:
Hm. I am unsure whether I like that. The next step here should be to eradicate `std::random` and use r123 everywhere. Then seeding should be a function of cell gid and the relevant item's index. For example one would seed an `event_generator` with `gid*100 + lid` where `lid` is the running count of the generator in the list. We might consider giving users a convenience function/helper thing there to partially automate this.

In general, I refrain from this for reasons of reproducibility, since all kinds of ordering or reentrancy (word!) come into play.